src/libpgmoneta/json.c

```diff
@@ -738,7 +744,12 @@ fill_value(char* str, char* key, uint64_t* index, struct json* o)
       idx++;
       while (idx < len && str[idx] != '"')
       {
-         val = pgmoneta_append_char(val, str[idx++]);
+         char ec_ch;
+         if (handle_escape_char(str, &idx, len, &ec_ch))
```

Comment:
I'm not quite sure about handling non-escaped characters inside the function as well. It's a little ambiguous, like you are treating normal `ch` as `ec_ch` as well. 

I still think it's better to only go into this handler when you encounter the backslash.

Reply:
And you get rid of the `escaped` flag this way.

Reply:
> I'm not quite sure about handling non-escaped characters inside the function as well. It's a little ambiguous, like you are treating normal `ch` as `ec_ch` nontheless.
> 
> I still think it's better to only go into this handler when you encounter the backslash.

Okay, I think maybe we can just change the name of the function if it is misleading. we can make it something like ‘get_next_char’

Reply:
> And you get rid of the `escaped` flag this way.

the main idea of escape flag was to detect any illegal use case.

Reply:
> Okay, I think maybe we can just change the name of the function if it is misleading. we can make it something like ‘get_next_char’

Yeah, but in terms of helper functions I think it's better to have their scopes smaller, instead of something that does a little bit of everything.

> the main idea of escape flag was to detect any illegal use case.

Well, you can still do that without it, If you only go into function when the condition is met, right?. The code would be cleaner.

Reply:
> > Okay, I think maybe we can just change the name of the function if it is misleading. we can make it something like ‘get_next_char’
> 
> Yeah, but in terms of helper functions I think it's better to have their scopes smaller, instead of something that does a little bit of everything.
> 
> > the main idea of escape flag was to detect any illegal use case.
> 
> Well, you can still do that without it, If you only go into function when the condition is met, right?. The code would be cleaner.


Okay on it.